Automattic/woocommerce-payments
# Address various platform checkout UI issues

Fixes #3856 

#### Changes proposed in this Pull Request

This PR addresses the following issues:
- Capture the phone number from the main checkout fields `billing_phone` (if it exists) and pre-fill that value into our phone number field.
- Block checkout form submission if our `Save my information...` box is checked but the phone number entered is invalid.
- Run validation provided by the `intl-tel-input` on the phone number entered.
- Ensure that the full phone number (including country code) is being submitted with the rest of the form data.
- Check that the email field is present before attempting to do anything with it.
- Validate the email address consistent with `is_email()` in WordPress before pinging the platform checkout to check the user.

#### Testing instructions

- Checkout this PR.
- Run `npm start`.
- Add product to cart on your local merchant site.
- Go to shortcode checkout.
- **Email Verification:**
   - Enter in `a@a` and verify that the email **IS NOT** checked on the platform checkout.
   - Enter in `a@a.` and verify that the email **IS NOT** checked on the platform checkout.
   - Enter in `a@a.c` and verify that the email **IS NOT** checked on the platform checkout.
   - Enter in `a@a.co` and verify that the email **IS** checked on the platform checkout.
- **Prefill Phone Number:**
   - Enter a valid phone number in the main checkout fields.
   - Check the "Save my information for faster checkouts." box.
   - Ensure that the phone number is copied to the new phone number field.
- **Phone Number Validation:**
   - Change the phone number to an invalid number `1234` and click out of the field.
   - Ensure that the field is marked with an error with helper text beneath.
   - Ensure that the "Place Order" button is disabled.
   - Re-enter a valid phone number and click out of the field.
   - Ensure that the "Place Order" button is enabled.
- **Checkout `$_POST` data:**
   - Fill out all necessary checkout data (email, phone, address, payment info)
   - Check the "Save my information for faster checkouts." box.
   - Submit checkout form.
   - Check the posted data to make sure it includes `$_POST['platform_checkout_user_phone_field']['full']` and that it includes the country code with the phone number entered ex. `+12124567890`
- **No javascript error on the order received page.**
   - After submitting your order, make sure there's no javascript error coming from `email-input-iframe.js`.
